core/src/main/java/org/elasticsearch/action/support/single/instance/TransportInstanceSingleOperationAction.java

```diff
@@ -169,11 +173,11 @@ protected boolean doStart() {
 
             if (!shard.active()) {
                 retry(null);
-                return false;
+                return;
             }
 
             if (!operationStarted.compareAndSet(false, true)) {
```

Comment:
I am not sure. I thought it was an additional protection against duplicated messages? Say for example, handleException() in line 203 is called twice for whatever reason then we might end up updating twice?